### PR TITLE
Angular Generator Upgrade (#694)

### PR DESCRIPTION
**Pull request by opiepj**
#694 Angular Generator Upgrade
## Checklist
 - [x]  A link to the issue/user story that the pull request relates to
 - [x]  How to recreate the problem without the fix
 - [x]  Design of the fix
 - [x]  How to prove that the fix works
 - [ ]  Automated tests that prove the fix keeps on working
 - [ ]  Documentation - any JSDoc, website, or Stackoverflow answers?

## Steps to Reproduce

All assuming I had the same network up
1. git clone https://github.com/opiepj/fabric-composer.git 
2. lerna bootstrap
3. cd fabric-composer/packages/generator-fabric-composer/generators/angular
4. yo ./index.js
5. cd generated app
6. npm start

## Existing issues

- [x] [StackOverflow issues](http://stackoverflow.com/tags/fabric-composer)
- [x] [GitHub Issues](https://github.com/fabric-composer/fabric-composer/issues)
- [x] [Rocket Chat history](https://chat.hyperledger.org/channel/fabric-composer)

## Design of the fix

Took an existing scaffold of the angular app, made modifications and copied it over to the templates, mostly best practices, updating modules, etc. Reran the scaffold to make sure everything was all right.

### CODE REVIEW

File: packages/generator-fabric-composer/generators/angular/templates/src/app/app-routing.module.ts [hunk 1]
```diff
@@ -6,6 +6,7 @@ import { <%= assetComponentNames[x] %> } from './<%= assetList[x].name %>/<%= as
 
 const routes: Routes = [
     // { path: 'transaction', component: TransactionComponent },
+    {path: '', redirectTo: '<%= assetComponentNames[0] %>', pathMatch: 'full'},
```
Comment (Jakeeyturner) [inline packages/generator-fabric-composer/generators/angular/templates/src/app/app-routing.module.ts]: Shouldn't this just return to the app component?

File: packages/generator-fabric-composer/generators/angular/templates/src/app/asset/asset.component.ts [hunk 2]
```diff
@@ -95,22 +101,24 @@ export class <%= currentAsset.name %>Component implements OnInit {
...
+    return this.service<%= currentAsset.name %>.deleteAsset(this.currentId).toPromise();
   }
 
-  setId(id:any):void{
+  setId(id: any): void{
     this.currentId = id;
   }
 
-  getForm(id:any):Promise<any>{
+  getForm(id: gany): Promise<any>{
```
Comment (opiepj) [inline packages/generator-fabric-composer/generators/angular/templates/src/app/asset/asset.component.ts]: Oops

File: packages/generator-fabric-composer/generators/angular/templates/src/app/app-routing.module.ts (hunk 1 above)
Comment (opiepj) [inline packages/generator-fabric-composer/generators/angular/templates/src/app/app-routing.module.ts]: The app component would be the parent with the `router-outlet` we need to default to at least one generated page, alternatively we can create a home component

**Comment by dselman** [comment]
What is the status of this. Can it be merged?

**Comment by opiepj** [comment]
@dselman going to rerun the CI build, but should be all set to merge from my end. @Jakeeyturner did you want me to take action on inserting an app component or is the default route enough?

**Comment by opiepj** [comment]
Is this good to merge into?

**Comment by sstone1** [comment]
@opiepj apologies, it's holiday season in the UK and we have no UI or Angular people in at the moment. Since you're happy to handle the home page in a separate PR, I'll merge this and let you and @Jakeeyturner decide if we need a new home page component.

Thanks for submitting this!

**Comment by opiepj** [comment]
@sstone1 Thank you!

**Pull request merged**